contracts/token/redeemable/token/RCSTKToken.sol

```diff
@@ -237,109 +259,111 @@ contract RCSTKToken is
+                SafeMath.mul(amountTokens, _iterations[_iteration].denominator),
+                _iterations[_iteration].numerator
             );
 
             emit MaximumTrustReached(msg.sender);
         }
 
         if (
             _amountDAI >=
-            iterations[_iteration].hardCap -
-                iterations[_iteration].totalReceived
+            _iterations[_iteration].hardCap -
+                _iterations[_iteration].totalReceived
         ) {
             /// @dev donate only up to the hardcap.
-            _amountDAI = SafeMath.sub(iterations[_iteration].hardCap, iterations[_iteration].totalReceived);
+            _amountDAI = SafeMath.sub(
+                _iterations[_iteration].hardCap,
+                _iterations[_iteration].totalReceived
+            );
 
             amountTokens = SafeMath.div(
-                SafeMath.mul(_amountDAI, iterations[_iteration].numerator),
-                iterations[_iteration].denominator
+                SafeMath.mul(_amountDAI, _iterations[_iteration].numerator),
+                _iterations[_iteration].denominator
             );
         }
 
         if (_amountDAI != 0) {
             bank.deposit(msg.sender, _amountDAI);
-            iterations[_iteration].totalReceived = SafeMath.add(
-                iterations[_iteration].totalReceived,
+            _iterations[_iteration].totalReceived = SafeMath.add(
+                _iterations[_iteration].totalReceived,
                 _amountDAI
             );
 
             if (
-                iterations[_iteration].totalReceived >
-                iterations[_iteration].softCap &&
-                iterations[_iteration].softCapTimestamp == 0
+                _iterations[_iteration].totalReceived >
+                _iterations[_iteration].softCap &&
+                _iterations[_iteration].softCapTimestamp == 0
             ) {
                 /// @dev Soft Cap is reached.
-                iterations[_iteration].softCapTimestamp = block.number;
+                _iterations[_iteration].softCapTimestamp = block.number;
                 bank.storeAllInVault();
                 emit SoftCapReached(_iteration);
             }
 
-            if (iterations[_iteration].softCapTimestamp == 0) {
+            if (_iterations[_iteration].softCapTimestamp == 0) {
                 _mint(msg.sender, amountTokens);
             } else {
                 /// @dev If softCap was reached we directly mint CSTK tokens and store donation into the Vault.
                 cstkTokenManager.mint(msg.sender, amountTokens);
                 bank.storeInVault(msg.sender, _amountDAI);
-            }/// TODO... I think it is fine that this is >= but it is kinda weird... what is the case that total receiveid is greater than hard cap... ?
+            }
             if (
-                iterations[_iteration].totalReceived >=
-                iterations[_iteration].hardCap
+                _iterations[_iteration].totalReceived ==
```

Comment:
We could leave `>=` as a sentinel (in case of overflow). But in principle, it would be better to make sure that case can never happen.